keras_nlp/samplers/beam_sampler.py

```diff
@@ -47,17 +47,18 @@ class BeamSampler(Sampler):
     char_lookup = {v: k for k, v in int_lookup.items()}
     batch_size, length, vocab_size = 1, 12, len(int_lookup)
 
-    def next(prompt, state, index):
+    def next(prompt, cache, index):
+        prompt_batch_size = tf.shape(prompt)[0]
+        hidden_states = tf.ones((prompt_batch_size, 1, 10))
```

Comment:
why is this 3d? I know it is unused in this case, but I think we could just do

```python
# A single float for our hidden state.
hidden_states = tf.zeros((prompt_batch_size, 1))
# A uniform distribution over our alphabet.
logits = tf.ones((prompt_batch_size, vocab_size))
```

Reply:
In our actual workflow (`call_with_cache`), the returned `hidden_states` is 3D.

Reply:
just saw your comment below, yea we can squeeze it to unify.